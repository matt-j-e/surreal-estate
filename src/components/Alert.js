import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Success = styled.p`
  padding: 1rem;
  text-align: center;
  color: mediumseagreen;
  border: 2px solid mediumseagreen;
  font-weight: 900;
`;

const Error = styled.div`
  padding: 1rem;
  text-align: center;
  color: palevioletred;
  border: 2px solid palevioletred;
  font-weight: 900;
`;

const Alert = ({ message, success }) => {
  if (!message) return "";
  return success ? <Success>{message}</Success> : <Error>{message}</Error>;
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
