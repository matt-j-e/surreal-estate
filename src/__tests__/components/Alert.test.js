import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../../components/Alert";

describe("Alert", () => {
  it("renders an Error alert", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(getByText("Error!").textContent).toBe("Error!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a 'Success' alert", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!!!" success />
    );
    expect(getByText("Success!!!").textContent).toBe("Success!!!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders nothing when no message to show", () => {
    const { asFragment } = render(<Alert message="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
