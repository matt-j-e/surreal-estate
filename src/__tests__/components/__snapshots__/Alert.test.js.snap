// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders a 'Success' alert 1`] = `
<DocumentFragment>
  <p
    class="sc-bdvvaa feCLug"
  >
    Success!!!
  </p>
</DocumentFragment>
`;

exports[`Alert renders an Error alert 1`] = `
<DocumentFragment>
  <div
    class="sc-gsDJrp jSTqYq"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert renders nothing when no message to show 1`] = `<DocumentFragment />`;
